skyllh/core/analysis.py

```diff
@@ -1591,3 +1608,231 @@ def initialize_trial(self, events_list, n_events_list=None, tl=None):
+        
+        Parameters
+        ----------
+
+        fitparams : dictionary with value for gamma, e.g. {'gamma': 2}
+        n : how many gaussians flares we are looking for
+        tol : the stopping criteria for expectation maximization. This is the difference in the normalized likelihood over the
+            last 20 iterations
+        iter_max : the maximum number of iterations, even if stopping criteria tolerance (tol) is not yet reached
+        sob_thres : set a minimum threshold for signal over background ratios. ratios below this threshold will be removed
+        initial_width : starting width for the gaussian flare in days
+        
+        Returns
+        -------
+        mean flare time, flare width, normalization factor for time pdf
+        
+        """
+        
+        ratio = self.get_energy_spatial_signal_over_backround(fitparams)
+        time = self._tdm_list[0].get_data("time")
+        
+        if sob_thresh > 0: # remove events below threshold
+            for i in range(len(ratio)):
+                mask = ratio > sob_thresh
+                ratio[i] = ratio[i][mask]
+                time[i] = time[i][mask]
+                
+        # in case, remove event
+        if remove_time is not None:
+            mask = time == remove_time
+            ratio = ratio[~mask]
+            time = time[~mask]
+
+        # expectation maximization
+        mu = np.linspace(self._data_list[0].grl["start"][0], self._data_list[-1].grl["stop"][-1], n+2)[1:-1]
+        sigma = np.ones(n) * initial_width
+        ns = np.ones(n) * 10
+        llh_diff = 100
+        llh_old = 0
+        llh_diff_list = [100] * 20
+
+        iteration = 0
+
+        while iteration < iter_max and llh_diff > tol: # run until convergence or maximum number of iterations
+            iteration += 1
+
+            e, logllh = expectation_em(ns, mu, sigma, time, ratio)
+
+            llh_new = np.sum(logllh)
+            tmp_diff = np.abs(llh_old - llh_new) / llh_new
+            llh_diff_list = llh_diff_list[:-1]
+            llh_diff_list.insert(0, tmp_diff)
+            llh_diff = np.max(llh_diff_list)
+            llh_old = llh_new
+            mu, sigma, ns = maximization_em(e, time)
+
+        return mu, sigma, ns
+
+
+    def run_gamma_scan_single_flare(self, remove_time=None, gamma_min=1, gamma_max=5, n_gamma=51):
+        """ run em for different gammas in the signal energy pdf
+        
+        Parameters
+        ----------
+        
+        remove_time : time information of event that should be removed 
+        gamma_min : lower bound for gamma scan
+        gamma_max : upper bound for gamma scan
+        n_gamma : number of steps for gamma scan
+        
+        Returns 
+        -------
+        array with "gamma", "mu", "sigma", and scaling factor for flare "ns_em"
+        """
+        
+        dtype = [("gamma", "f8"), ("mu", "f8"), ("sigma", "f8"), ("ns_em", "f8")]
+        results = np.empty(51, dtype=dtype)
```

Comment:
Index out of bounds bug fix.
```suggestion
        results = np.empty(n_gamma, dtype=dtype)
```